lib/connections/metrics.go

```diff
@@ -18,8 +20,30 @@ var metricDeviceActiveConnections = promauto.NewGaugeVec(prometheus.GaugeOpts{
 	Help:      "Number of currently active connections, per device. If value is 0, the device is disconnected.",
 }, []string{"device"})
 
-func registerDeviceMetrics(deviceID string) {
+func registerDeviceMetrics(dc config.DeviceConfiguration) {
+	registerDeviceInfoGauge(dc)
 	// Register metrics for this device, so that counters & gauges are present even
 	// when zero.
+	deviceID := dc.DeviceID.String()
 	metricDeviceActiveConnections.WithLabelValues(deviceID)
 }
+
+func registerDeviceInfoGauge(dc config.DeviceConfiguration) {
+	// Create a dynamic "info" gauge to help users
+	// map IDs to human-readable strings.
+	// It produces a constant `1`
+	did := dc.DeviceID.String()
+	info_gauge := prometheus.NewGaugeFunc(
+		prometheus.GaugeOpts{
+			Namespace: "syncthing",
+			Name:      "device_info_" + did,
```

Comment:
Generally one wouldn't do this as separate metrics, but one metric with multiple label values, e.g.

```
syncthing_device_info{id="aaa", name="bbb"} = 1
syncthing_device_info{id="ccc", name="ddd"} = 1
```

etc, otherwise it will be very difficult to use in joins I think. Probably the easiest way to do this would be to have a custom collector that has the config in entirety and generates the metrics when requested to

Reply:
Ok, cool, I will look into a custom collector.

Is it expected that most syncthing users will be using Grafana for observability monitoring?  As I progress, I would like to base my ergonomic/usability testing on whatever is most likely to benefit actual users.

Reply:
I'm not sure what you're asking really; I expect a very tiny minority of Syncthing users to use any form of observability. Of those that do, I think Grafana on either Prometheus or VictoriaMetrics etc is the absolutely most likely combination, yes.

Reply:
You've given me enough on which to work.  I want to recreate a "common" setup and confirm that my final code "works" ergonomically to expose the information intended.

Sorry for the delay in replying.  We had a death in the family which has kept me busy throughout this week.  I expect to be able to work on a custom collector early next week.